src/sources/urlresource.h

```diff
@@ -19,11 +19,16 @@ class UrlResource {
     }
 
 protected:
-    explicit UrlResource(QUrl url)
+    explicit UrlResource(const QUrl& url)
         : m_url(url) {
     }
 
+    inline bool isLocalFile() const {
+        return getUrl().isLocalFile();
+    }
+
     inline QString getLocalFileName() const {
+        DEBUG_ASSERT(isLocalFile());
```

Comment:
What happens on windows with network shares? 
There is a pending bug for SMB locations 
https://bugs.launchpad.net/mixxx/+bug/918222
Will this fail here?